swift/BUILD.bazel

```diff
@@ -0,0 +1,65 @@
+load("@rules_pkg//:mappings.bzl", "pkg_attributes", "pkg_filegroup", "pkg_files")
+load("@rules_pkg//:install.bzl", "pkg_install")
+load("@ql//:defs.bzl", "codeql_platform")
+load("@ql_utils//:source_dir.bzl", "current_source_dir")
+
+pkg_files(
+    name = "dbscheme",
+    srcs = [
+        "ql/lib/swift.dbscheme",
+        "ql/lib/swift.dbscheme.stats",
+    ],
+)
+
+pkg_files(
+    name = "qltest",
+    srcs = ["tools/qltest.sh"],
+    attributes = pkg_attributes(mode = "0755"),
+    prefix = "tools",
+)
+
+pkg_files(
+    name = "manifest",
+    srcs = ["codeql-extractor.yml"],
+)
+
+pkg_filegroup(
+    name = "extractor-pack-generic",
+    srcs = [
+        ":dbscheme",
+        ":manifest",
+        ":qltest",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+pkg_files(
+    name = "extractor",
+    srcs = ["//swift/extractor"],
+    attributes = pkg_attributes(mode = "0755"),
+    prefix = "tools/" + codeql_platform,
+)
+
+pkg_filegroup(
+    name = "extractor-pack-arch",
+    srcs = [":extractor"],
+    visibility = ["//visibility:public"],
+)
+
+pkg_filegroup(
+    name = "extractor-pack",
+    srcs = [
+        ":extractor-pack-arch",
+        ":extractor-pack-generic",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+pkg_install(
+    name = "create-extractor-pack",
+    srcs = ["//swift:extractor-pack"],
+    args = [
+        "--destdir",
+        current_source_dir() + "/extractor-pack",
```

Comment:
I wonder if we should move this from the hermeticity-breaking `current_source_dir` rule to the documentation instead.
That would allow us to get rid of the extra repository, and of somebody accidentally using `current_source_dir` anywhere else.
What do you think?

Reply:
yeah, I was also a bit wary of this, then decided to go with what was done in ruby. We could skip the `--destdir` and point `.clodeqlmanifest` to the corresponding place in the target's runfiles (as we do not need to worry about window here)...

Reply:
Can you explain what you mean with "what was done in ruby"? I have not looked at their extractor build at all.

Reply:
they have a `create-extractor-pack.sh` script that when run without any arguments will instantiate the extractor pack in `ruby/extractor-pack`, which in turn is referenced by `ruby/.codeqlmanifest.json`.

Any way, I found a more elegant solution by wrapping the install script setting `DESTDIR` using `BUILD_WORKSPACE_DIRECTORY', so the scary `current_source_dir()` can go away :slightly_smiling_face:

Reply:
(which by the way allows cleaning up the pack directory, which might otherwise pose problems with leftover files)

Reply:
Yeah that's the reason we do the whole delete+unzip in sembuild all the time - having a good solution that's a bit faster would be a good model for something to adopt there, too :-)